### PR TITLE
Add S3 path style access support to exchange spooling

### PR DESCRIPTION
**Pull request by linzebing**
## Description
As title

## Non-technical explanation
As title

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
() Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Review by losipiuk — APPROVED** [review]

**Comment by losipiuk** [comment]
@linzebing please update PR desc to include release notes text.

**Comment by colebow** [comment]
If we're adding a new configuration property, we need to make sure it gets documented. I'll get up a PR before release, but please try to include docs with the initial PR.

cc @losipiuk

**Comment by colebow** [comment]
There are a few reasons why a thing may be undocumented:

1. We messed up and it should've been documented (this is common - especially for older features that landed before we began making a push for better documentation).
2. It's experimental and it's going to be documented once it's better-tested and ready for wide use.
3. We're worried that documenting it will give users the bright idea to fiddle with something that they really shouldn't.

Not-so-great reasons to skip docs on something:

-  Most people will use the default.
-  It's a niche setting / not very important.

So basically, we only shouldn't document something when we're worried that telling our users about it is going to run the risk of causing problems. If the only consequence of documenting it is that it takes up a little extra space, we should put it in the docs.
